src/core/core.scale.js

```diff
@@ -347,26 +348,49 @@ class Scale extends Element {
 		return null;
 	}
 
+	/**
+	 * @private
+	 * @since 3.0
+	 */
+	_getUserBounds() {
+		var min = this._userMin;
+		var max = this._userMax;
+		if (isNullOrUndef(min) || isNaN(min)) {
+			min = Number.POSITIVE_INFINITY;
+		}
+		if (isNullOrUndef(max) || isNaN(max)) {
+			max = Number.NEGATIVE_INFINITY;
+		}
+		return {min, max, minDefined: isFinite(min), maxDefined: isFinite(max)};
+	}
+
+	/**
+	 * @private
+	 * @since 3.0
+	 */
 	_getMinMax(canStack) {
 		var me = this;
-		var metas = me._getMatchingVisibleMetas();
-		var min = Number.POSITIVE_INFINITY;
-		var max = Number.NEGATIVE_INFINITY;
+		var {min, max, minDefined, maxDefined} = me._getUserBounds();
 		var minPositive = Number.POSITIVE_INFINITY;
-		var i, ilen, minmax;
+		var i, ilen, metas, minmax;
+
+		if (minDefined && maxDefined) {
+			return {min, max};
+		}
 
+		metas = me._getMatchingVisibleMetas();
 		for (i = 0, ilen = metas.length; i < ilen; ++i) {
 			minmax = metas[i].controller._getMinMax(me, canStack);
-			min = Math.min(min, minmax.min);
-			max = Math.max(max, minmax.max);
+			if (!minDefined) {
```

Comment:
You could probably make this slightly more efficient by just taking the `min` and `max` here regardless and then doing the check for the user-provided value at the end when you return

Reply:
I don't think its worth the effort, the loop is around visible datasets - not going to have thousands of those in any usable scenario.